packages/qri-node/lib/qri-node.js

```diff
@@ -34,27 +50,86 @@ module.exports = {
             command += ` --file ${filePath}`
           })
         }
-        const child = spawn(command, [], { shell: true, stdio: [process.stdin, process.stdout, process.stderr] })
-        child.on('exit', () => {
-          console.log('done!')
-          resolve()
-        })
+        const { status, stdout, stderr } = spawnSync(command, [], { shell: true })
+
+        if (status === 1) {
+          reject(new Error(stderr.toString().trim()))
+        } else {
+          resolve(stdout.toString().trim())
+        }
+      } catch(e) {
+        reject(e)
+      }
+    })
+  },
+
+  /**
+  * Get a dataset or a specific component from a dataset
+  * @param {string} dsRef - The dataset reference to save
+  * @param {string} component - The component to get (body, readme, structure, meta)
+  * @returns {Promise} - If resolved, stdout, if rejected, stderr
+  */
+  get: (dsRef, component = '') => {
+    console.log(`getting qri dataset ${dsRef}`)
+    return new Promise((resolve, reject) => {
+      try {
+        let command = `qri get ${component} ${dsRef}`
+
+        const { status, stdout, stderr, output } = spawnSync(command, [], { shell: true })
+        if (status === 1) {
+          reject(new Error(stderr.toString().trim()))
+        } else {
+          resolve(stdout.toString().trim())
+        }
+      } catch(e) {
+        reject(e)
+      }
+    })
+  },
 
+  /**
+  * Remove a Dataset
+  * @param {string} dsRef - The dataset reference to save
+  * @param {Object} removeOptions
+  * @param {boolean} removeOptions.all whether to remove all versions
+  * @returns {Promise} - If resolved, stdout, if rejected, stderr
+  */
+  remove: (dsRef, { all = false }) => {
+    console.log(`removing qri dataset ${dsRef}`)
+    return new Promise((resolve, reject) => {
+      try {
+        let command = `qri remove ${dsRef}`
+        if (all) {
```

Comment:
Without `all`, the command needs to specify how many versions should be removed. If neither `--all` nor `--versions` are given, the command will fail.

Reply:
will fix